### ISSUE TITLE
Doesn't work with `/examples` folder

### ISSUE THREAD

**Issue by setoelkahfi** [rust-lang/log#647]
Hello, this log doesn't seem to work with the [`Cargo` examples](https://doc.rust-lang.org/cargo/reference/cargo-targets.html#examples) directory. Is it by design?

I have a `basic.rs` file inside my examples folder crate. This doesn't print the log information:
```rust
use std::env;
use log::{debug, error, info};

#[tokio::main]
async fn main() {
  debug!("Try to run the example");
  // Some other lines...
```

This one does:
```rust
use std::env;
use log::{debug, error, info};

#[tokio::main]
async fn main() {
  println!("Try to run the example");
  // Some other lines...
```

I ran that example with: `cargo run --example basic`

**Comment by Thomasdezeeuw** [comment]
@setoelkahfi the log crate only provides a logging facade, you'll also need an implementation that does the actual logging. See one of the implementation in https://github.com/rust-lang/log?tab=readme-ov-file#in-executables.

**Comment by setoelkahfi** [comment]
@Thomasdezeeuw I'm building a library. And per the [readme](https://github.com/rust-lang/log?tab=readme-ov-file#in-libraries), I should only link the log crate.

**Comment by Thomasdezeeuw** [comment]
> [@Thomasdezeeuw](https://github.com/Thomasdezeeuw) I'm building a library. And per the [readme](https://github.com/rust-lang/log?tab=readme-ov-file#in-libraries), I should only link the log crate.

Yes. Then you can add an implementation as `dev-dependency` (https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#development-dependencies), which you can use in the examples, but it will not get added to your library.

**Comment by setoelkahfi** [comment]
I don't think I got this. I tried to use the env_logger one, for example, I tried to init it in my lib.rs file, but it says `Syntax Error: expected BANGrust-analyzer[syntax-error](https://doc.rust-lang.org/stable/reference/)`

```rust
use crate::model::{ErrorDetail, ErrorResponseCode, Link, SessionError};
use log::{debug, info};
use reqwest::{Client, Response, StatusCode};

env_logger::init(); // Can't do
```

Do I need to specify a specific feature?

**Comment by KodrAus** [comment]
Just coming back through some triage. There hasn't been any activity here for a while so I'll go ahead and close this one. If you're running into trouble with Rust there are also some community forums with people who can help you get unstuck: https://www.rust-lang.org/community
